### DISCUSSION TITLE
Bug When Combined With Hyperkey

### DISCUSSION THREAD

**Discussion by RyanSharafuddin** [rxhanson/Charmstone-Community#28] [Q&A] [answered]
Love this program. I think I've encountered a bug, though. I've used Hyperkey to set caps lock to be the hyperkey (all 4 of shift, control, option, and command). I then used the Charmstone settings to activate on pressing those 4 keys and the arrow keys/mouse. When I press caps lock + arrow key, this works as expected. However, when I do caps lock + move mouse, the app icons don't show up. They do show up when I press down all 4 of the modifier keys + move mouse, though.

**Comment by rxhanson** [comment, marked as answer]
Thanks for reporting. In Hyperkey, do you have "Apply modifiers to keypress events and" **move & drag** checked?

**Reply by RyanSharafuddin** [reply]
Ahh, I did not. I checked those settings, and now it works. Thanks!

**Comment by xemza** [comment]
This doesn't work for me when I have caps lock as the hyper key. Well, ⇪ + arrow keys works, but ⇪ + mouse movement does not. However, when I change the hyper key to any other key like option or command, everything works as expected. Any Ideas?

**Reply by rxhanson** [reply]
Hmm. Honestly, I'm not sure how this could be the case. Maybe send a couple quick screenshots of your settings windows, as a quick sanity check?

**Reply by xemza** [reply]
Ok, it's fixed now. The problem was that I had caps lock set to "no action" in macOS keyboard settings. I turned it back on and now moving and dragging work as well. Interesting that keypress events work regardless. Anyway it's all good now and I appreciate the quick reply!
